File: 2753-count-houses-in-a-circular-street-ii/2753-count-houses-in-a-circular-street-ii.js
/**
 * Definition for a street.
 * class Street {
 *     @param {number[]} doors
 *     constructor(doors);
 * 
 *     @return {void}
 *     closeDoor();
 * 
 *     @return {boolean}
 *     isDoorOpen();
 * 
 *     @return {void}
 *     moveRight();
 * }
 */
/**
 * @param {Street} street
 * @param {number} k
 * @return {number}
 */
var houseCount = function(street, k) {
    while(!street.isDoorOpen()) {
        street.moveRight();
    }
    let ans = 0;
    for(let i = 1; i <= k; ++i) {
        street.moveRight();
        if(street.isDoorOpen()) {
            ans = i;
            street.closeDoor();
        }
    }
    return ans;
};

File: 2753-count-houses-in-a-circular-street-ii/2753-count-houses-in-a-circular-street-ii.go
/**
 * Definition for a street.
 * type Street interface {
 *     CloseDoor()
 *     IsDoorOpen() bool
 *     MoveRight()
 * }
 */
func houseCount(street Street, k int) int {
    for !street.IsDoorOpen() {
        street.MoveRight()
    }
    var ans int = 0
    for i:=1; i<=k; i++ {
        street.MoveRight()
        if street.IsDoorOpen() {
            ans = i;
            street.CloseDoor()
        }
    }
    return ans
}